opacus/accountants/gdp.py

```diff
@@ -3,25 +3,31 @@
 
 
 class GaussianAccountant(IAccountant):
-    def __init__(self, noise_multiplier, sample_rate, poisson):
-        self.noise_multiplier = noise_multiplier
-        self.sample_rate = sample_rate
-        self.poisson = poisson
+    def __init__(self):
+        self.noise_multiplier = None
+        self.sample_rate = None
         self.steps = 0
 
     def step(self, noise_multiplier: float, sample_rate: float):
+        if self.noise_multiplier is None:
+            self.noise_multiplier = noise_multiplier
+
+        if self.sample_rate is None:
+            self.sample_rate = sample_rate
+
         if noise_multiplier != self.noise_multiplier or sample_rate != self.sample_rate:
             raise ValueError(
                 "Noise multiplier and sample rate have to stay constant in GaussianAccountant."
             )
         self.steps += 1
 
-    def get_privacy_spent(self, delta: float) -> float:
-        if self.poisson:
+    def get_epsilon(self, delta: float, poisson: bool = True) -> float:
+        if poisson:
```

Comment:
Why move poisson here?

Reply:
Got it actually! Seems necessary to not have any extra parameters in the constructor